kernel/rtlil.h

```diff
@@ -977,6 +979,8 @@ struct RTLIL::Module : public RTLIL::AttrObject
 
 	RTLIL::IdString name;
 	pool<RTLIL::IdString> avail_parameters;
+	dict<RTLIL::IdString, RTLIL::ParameterInfo> parameter_information;
+	dict<RTLIL::IdString, dict<RTLIL::IdString,RTLIL::Const>> parameter_attributes;
```

Comment:
Every RTLIL class is derived from `RTLIL::AttrObject` which holds attributes in a `dict<RTLIL::IdString, RTLIL::Const>`. Since we do not want a RTLIL object for parameters without  attributes I declared such a dict directly here. Parameters without attributes are not present in it.